taipy/core/data/_data_manager.py

```diff
@@ -75,9 +74,9 @@ def _can_create(cls, config: Optional[DataNodeConfig] = None) -> Reasons:
 
         if config is not None:
             if not isinstance(config, DataNodeConfig):
-                reason._add_reason(config_id, _build_wrong_config_type_reason(config_id, "DataNodeConfig"))
+                reason._add_reason(config_id, WrongConfigType(config_id, DataNodeConfig.__name__))
```

Comment:
Since the name difference between `Reasons` and `Reason` classes is only singular or plural, I would love it if the variables to match the class name to avoid future confusion.

```suggestion
                reasons._add_reason(config_id, WrongConfigType(config_id, DataNodeConfig.__name__))
```